plugins: vision: use complete visual odometry msg to send to the FCU

TSC21 opened the pull request:
Purpose: add a more complete simulated odometry data sent from Gazebo to the FCU.

potaito committed:
tap_esc: converted to module

dagar committed:
move pressure altitude from baros to sensors module

acfloria committed:
En-/disable transmitting from a mavlink instance using a uORB message

Allow enabling/disabling transmitting data from all mavlink instances
with a specific mode/type. When disabled the instance can still receive
data. The uORB message used is the vehicle_command and uses the following
parameter:
param1: Specifies the mode/type of the instances to enable/disable transmitting
param2: Boolean to indicate if transmitting should be enabled or disabled

priseborough committed:
simulator: Use baro pressure direct from simulator

acfloria committed:
Improvements for the iridiumsbd driver

- Update heartbeat after successful sbd session and fill it in the telemetry_status message
- Add parameter for session timeout, read interval, and stacking time
- Add sbd session timeout
- Fix updating the rssi at a constant rate when no sbd session is scheduled
- Add variable timout to read_at function
- Check if test command is valid to avoid being stuck in the test state

acfloria committed:
Add switch logic for high latency telemetry in commander

Switch to the high latency telemetry, if available, if every low latency link is lost. Switch back if any low latency link is regained. Only indicate that all links are lost if all high and low latency links are lost. Allow different timeouts for high and low latency links.

acfloria committed:
Refactor Iridiumsbd driver

- Reorder functions so that they have the same order in the .h and .cpp file
- Update documentation for functions
- Make variables and some functions private

acfloria committed:
Commander: High latency telem fallback only in non-manual mode

potaito committed:
snapdragon: added missing includes

potaito committed:
snapdragon: fixed compiler shadow warning, function name was identical as a variable name

potaito committed:
snapdragon: changed constexpr to const in FlightTask so that it would compile

potaito committed:
snapdragon: Applied camelCasing

acfloria committed:
Style fix and iridium signal check rate fix

acfloria committed:
Remove magic numbers from commander.cpp and style fixes

acfloria committed:
commander: Fix for qurt and improve high latency switch logic

Stifael committed:
smoothing xy: remove unused variable

Stifael committed:
FlightTaskUtility: remove duplicate slewrate

Stifael committed:
clang-tidy: don't check for pass-by-value optimization

MaEtUgR committed:
FlightTasks: switch field name from thr to thrust + some code spacing

MaEtUgR committed:
FlightTasks: create enumeration type for the task index while still offering integer index with checks

MaEtUgR committed:
FlightTaskManual: remove unnecessary activate method

MaEtUgR committed:
FlightTasks: corrected comment and sorted out unnecessary includes

MaEtUgR committed:
FlightTaskManualAltitude/Position: renamed and recommented velocity hold threshold

Stifael committed:
FlightTask: rebse fix for empty setpoint

acfloria committed:
Add the SimpleAnalyzer to generate messages for MAVLink streams

dagar committed:
commander add parameters for eph, epv, evh requirements

dagar committed:
FW defaults set commander eph, epv, evh thresholds

acfloria committed:
Add the HighLatency2 mavlink stream

acfloria committed:
Fix format and remove ets driver from fmu-v2 build

Stifael committed:
empty setpoint add NAN for thrust, add comment

acfloria committed:
Adopt mavlink stream to new HIGH_LATENCY2 definition

Stifael committed:
FlightTaskManualPosition: use axis angle to rotate into world frame

Stifael committed:
FlightTaskManualSmoothing: change name of method

Stifael committed:
FlightTaskManual: method to reset all members setpoints to NAN

Stifael committed:
FlightTaskManual: move all setpoints into base class

Stifael committed:
FlightTaskManualStabilized: update method that sets all member setpoints

Stifael committed:
FlightTaskManualAltitude: remove member setpoints and update method

Stifael committed:
FlightTaskManualAltitudeSmooth: remove unused member methods/variables

Stifael committed:
FlightTaskManualPosition: remove update method and old member variable setpoints

Stifael committed:
FlightTaskManualPositionSmooth: remove unused mehthods and member variables

Stifael committed:
FlightTaskSport: replace rotation by axis angle

Stifael committed:
FlightTaskManualSmoothingZ: remove empty space

Stifael committed:
FlightTaskManualStabilized: check for 0 maximum tilt and scale throttle linearly

Stifael committed:
ControlMath: reference to by value

Stifael committed:
FlightTaskSport: just scale velocity septoint

dagar committed:
move baro and magnetometer data out of sensor_combined

dagar committed:
commander update super block parameters

Commit:
ManualSmoothingXY: brake with large default jerk max

Commit:
PositionControl: parameter naming fix

Commit:
ControlMath: don't consider sign when vector length is 0

Commit:
FlightTaskManual: replace yaw with yaw_sp for rotation

Commit:
PositionControl: remove legacy yaw logic

Commit:
PositionControl: remove unused parameters

Commit:
FlightTaskManualStabilized: limit manual yaw rate in flighttask

Commit:
FlightTaskManual: transformation into heading frame use vehicle yaw or yaw_sp based
on vehicle rotation in yaw

Commit:
PositionControl: comment fix + unused parameter removal

Commit:
Improve the Crazyflie MAVLink tunnel to increase efficiency

This change fragments MAVLink packets more efficiently and therefore increases the net throughput. This in turn makes the connection significantly more stable and the Crazyflie experience overall more usable.

acfloria committed:
Fixes IridiumSBD driver bug when transmitting data

acfloria committed:
Adopt high latency switching to new MAV_CMD design

MaEtUgR committed:
Revert "Subscription remove unused instance class member"

This reverts commit cf2d794da9026dfe7ba2cd746e767fddaf4666e3.

MaEtUgR committed:
FlightTasks: add library to cmake configurations

all the ones with mc_pos_control need the library, we should use cmake dependencies!

MaEtUgR committed:
FlightTasks: comment out mavlink command processing

because upstream there needs to be a
common mavlink command definition first
and then it can be easily reenabled again

MaEtUgR committed:
mc_pos_control: refactor, move smooth takeoff velocity into a function

# Conflicts:
#	src/modules/mc_pos_control/mc_pos_control_main.cpp

MaEtUgR committed:
mc_pos_control: refactor, move landed thrust reduction into function

and make it use matrix library for flight task compatibility

# Conflicts:
#	src/modules/mc_pos_control/mc_pos_control_main.cpp

MaEtUgR committed:
mc_pos_control: refactor, indent control block

individual commit for the indentation because otherwise
the diff gets unreadable. this indentation is made because
afterwards the entire legacy position control functionality
is in an else case.

acfloria committed:
Set the HIGH_LATENCY2 stream to a constant rate

acfloria committed:
Do not enable hil for high latency mavlink

MaEtUgR committed:
mc_pos_control: enable flight tasks

instantiate flight tasks and the refactored position controller class
only use the new functionality if a temporary parameter is set for testing

MaEtUgR committed:
FlightTasks: replace setpoint setters with members

I realized that instead of using the setpoint setters inline
in real world tasks everyone started to have its own setpoint
member variable and only call the setter in the end for all the
privatly generate setpoints. This makes the setter useless and
therefore I switched to member setpoints in the architecture.
They bring more felxibility which is obviously needed but also
less structure which is the price to pay.

MaEtUgR committed:
FlightTasks: adapt tasks to member setpoints

The FlightTaskManual... tasks already had their internal
setpoint member variables. I switched them to use the
architecture with setpoint member variables as it was
implemented the commit before. They simplify a lot.

MaEtUgR committed:
FlightTaskManualStabilized: remove double limit of yawspeed

The same paramater was used twice, once for scaling the yaw stick
and once for limitng just after. This was useless.

Commit:
FlightTask: only allow for position and alitude control without smoothing.
This commit is only done to enable incremental testing of Flighttask.

MaEtUgR committed:
FlightTasks: fix manual mode takeoff

With the new position control architercture manual mode and
potition mode run through the same logic and as a result
it had a throttle thershold for smooth takeoff also in
manual mode. This is fixed by by ignoring the threshold for
any mode which doesn't control the climb rate.

acfloria committed:
Small improvements for the IridiumSBD driver

acfloria committed:
Transmit one message over high latency telemetry on boot

acfloria committed:
Update timestamp for last sent message im mavlink stream only if the message was sent

acfloria committed:
Only generate the HIGH_LATENCY2 message if transmitting is allowed

dagar committed:
replay update sensor_combined fields

Commit:
PositionControl: Anti-windup that uses max in NE based on tilt limit and remaining throttle
excess

Commit:
ControlMath: remove unused methods

acfloria committed:
Improve message handling in mavlink IRIDIUM mode

- Do not send any mission updates as they are handled within HIGH_LATENCY2
- Always send a command acknowledge even if transmitting is not enabled

dagar committed:
VTOL preserve PWM min and max when enabling/disabling actuators

acfloria committed:
Fix bug when ISBD parameters were too large

MaEtUgR committed:
PositionControl: refactor remove Data define for Vector

This define results in super unreadable code.

acfloria committed:
Reduce required stack for HIGH_LATENCY2

acfloria committed:
Remove exception for command acknowledge in mavlink transmitting

acfloria committed:
Improve IridiumSBD TX buffer handling

- Every time data is written to the iridium module all previous untransmitted data is overwritten
- The buffer is reset only on the following conditions
    - A sbd session is successfully completed
    - An incoming mavlink message would result in less than SATCOM_MIN_TX_BUF_SPACE free bytes after it is written
    - Any other incomming data would result in less than SATCOM_MIN_TX_BUF_SPACE free bytes after it is written to the buffer.

Commit:
PositionControl: replace matrix length() method with simple dot-product to safe compuation

acfloria committed:
Iridiumsbd driver: Publish log message when TX buffer is deleted

nanthony21 committed:
Use filtered throttle for battery estimation instead of raw.

nanthony21 committed:
Fix formatting

acfloria committed:
IridiumSBD: Only write to modem if new data arrived

acfloria committed:
IridiumSBD: Improve TX buffer handling

- Allow using the full buffer size (340 bytes)
- Add tx mutex for sbd session to avoid dropping message which are written during an sbd session

acfloria committed:
IridiumSBD: Do not allow stopping the driver when the device is used to prevent hardfault

acfloria committed:
MavlinkStream: Ensure that not multiple messages are sent after some time sending nothing

acfloria committed:
IridiumSBD: Robustify state machine and fix format

- Check if changing to a non standby state if the current state is the standy state and no change is already scheduled.
- Add prefix _ to all class variables

acfloria committed:
Increase the MavLink module stack size

dagar committed:
posix sitl ekf2 plane model add fw defaults

 - EKF2 airspeed fusion and synthetic sideslip fusion
 - commander eph/epv failsafe thresholds appropirate for FW

dagar committed:
commander move remaining position failsafe params to class and cleanup

dagar committed:
commander add COM_POS_FS_PROB and COM_POS_FS_DELAY metadata limit

acfloria committed:
Commander: Improve handling and generating of VEHICLE_CMD_CONTROL_HIGH_LATENCY

Commit:
PositionControl: thrust setpoint check with fabsf

RomanBapst committed:
vtol_att_control: merge methods to disable a selection of motors

- moved methods used to modify channel maximum pwm value to VtolType
class so that the can be shared by the differnt vtol types

Signed-off-by: Roman <bapstroman@gmail.com>

RomanBapst committed:
vtol_att_control: move handling of motor state and idle speed selection
to VtolType class

Signed-off-by: Roman <bapstroman@gmail.com>

RomanBapst committed:
vtol_type: improve comments

Signed-off-by: Roman <bapstroman@gmail.com>

dagar committed:
commander restore MC nav failure latch

dagar committed:
commander delete unused DIFFPRESS_TIMEOUT

dagar committed:
EKF2 and replay add all consumed messages to ekf2_timestamps and refactor

priseborough committed:
ekf2: Enable adjustment of nav validity timeout

RomanBapst committed:
vtol_type.h: initialise flag_idle_mc correctly

Signed-off-by: Roman <bapstroman@gmail.com>

RomanBapst committed:
vtol_att_control: initialise VT_FW_MOT_OFFID correcly for standard vtol

Signed-off-by: Roman <bapstroman@gmail.com>

acfloria committed:
Commander: Move telemetry logic to separate functions

bkueng committed:
mc_pos_control: refactor BlockParam -> Param

bkueng committed:
flight tasks: refactor BlockParam -> Param and handle param updates

bkueng committed:
flight tasks smoothing: fix style issues

bkueng committed:
flight tasks smoothing: refactor param_find -> Param

Plus some style fixes

RomanBapst committed:
drv_pwm_output: include board config header

- board config can define PX4_PWM_ALTERNATE_RANGES which needs to be known
by the compiler prior to processing this file

Signed-off-by: Roman <bapstroman@gmail.com>

RomanBapst committed:
vtol_type: avoid code duplication

- created one method which has access to the pwm output device

Signed-off-by: Roman <bapstroman@gmail.com>

bkueng committed:
px4_param.h: add override to _DEFINE_PARAMETER_UPDATE_METHOD_CUSTOM_PARENT

RomanBapst committed:
vtol_type.cpp: close pwm output device file descriptor

Signed-off-by: Roman <bapstroman@gmail.com>

acfloria committed:
Commander: Fix bug when changing to high latency telemetry

Due to the old heartbeat of the high latency telemetry when activating it after flying sufficiently long in normal telemetry it is first detected as lost until the first message is sent.
By updating the heartbeat to the current time on switching this issue is avoided.

Also includes a small style fix for the HIGH_LATENCY2 stream

acfloria committed:
revert v2 cmake changes

dagar committed:
commander state machine helper shorten names

dagar committed:
commander status flags delete condition_system_prearm_error_reported

 - this flag often results in hiding useful information, or adding
   useless information to the mavlink console

dagar committed:
commander arming state machine pass checks by default

 - this is needed for the unit tests

dagar committed:
commander prearm_check is always prearm

dagar committed:
commander move power check to prearm_check

dagar committed:
commander arming state transition delete useless errors

dagar committed:
commander move safety check to prearm

dagar committed:
commander move avionics rail voltage check out of state machine

 - add preflight_check helper

dagar committed:
commander set status_flags.usb_connected directly and delete static

dagar committed:
commander COMPONENT_ARM_DISARM remove useless arm error

dagar committed:
commander prearm_check limit simultaneous error messages

dagar committed:
commander arming state transition consolidate HIL special checks and remove atomic

dagar committed:
commander arming_state_transition cleanup preflight and prearm calls

 - only call prearm if preflight passes
 - prearm always provide feedback

dagar committed:
commander preflight sensor failure report one at a time

dagar committed:
commander move arm authorization to prearm_check

dagar committed:
commander state machine fix preflight and prearm error

 -  fixes #9155

dagar committed:
commander preflightCheck fail silently for the 2 seconds (rather than skip checks entirely)

dagar committed:
commander state machine add global position check

dagar committed:
commander arm_auth check last

dagar committed:
fw_pos_control_l1 don't store TECS and L1 parameters

acfloria committed:
Commander: Remove Mavlink from CMakeFile

dagar committed:
aerocore2 config remove telemetry modules, FW and GND controllers

RomanBapst committed:
FixedWingAttitudeControl: use trim airspeed if airspeed is disabled

- prior to this fix the fw attiude controller used airspeed to scale control
surfaces even though airspeed was disabled

Signed-off-by: Roman <bapstroman@gmail.com>

RomanBapst committed:
tailsitter: do not condition transition on airspeed if airspeed disabled

Signed-off-by: Roman <bapstroman@gmail.com>

acfloria committed:
Mavlink: Use filtered update rate for HIGH_LATENCY2 stream instead of main loop delay

dagar committed:
commander preflight checks pass status and status_flags messages

dagar committed:
commander preflightCheck don't report failures during calibration

Commit:
commander: Remove usb_ connected flag on battery warnings

Battery warning gets only fired if a battery is inserted.

acfloria committed:
Mavlink: Add acknowledge for VEHICLE_CMD_CONTROL_HIGH_LATENCY

CarlOlsson committed:
ll40ls: increase the number of samples used to find a correlation peak for LitarLite

Signed-off-by: CarlOlsson <carlolsson.co@gmail.com>

CarlOlsson committed:
ll40ls: increase the sleep time after resetting registers

Signed-off-by: CarlOlsson <carlolsson.co@gmail.com>

Commit:
explaining why we send 30 bytes crtp msg data instead of 31

acfloria committed:
Commander: Combine telemetry variables into struct

acfloria committed:
Remove unnecessary newlines

bkueng committed:
commander: immediately publish kill switch state change

acfloria committed:
IridiumSBD: Log tx buffer reset in the telemetry_status instead of a status text

acfloria committed:
IridiumSBD: Clean up MavLink message parsing

acfloria committed:
Move VEHICLE_CMD_CONTROL_HIGH_LATENCY ack to commander

okalachev committed:
Add new mode AUTO_PRECLAND

stmoon committed:
clarify the codes of LPE

dagar committed:
update px4fmu_test and add px4fmu-v2_test to Jenkins

dagar committed:
Update submodule mavlink v2.0 to latest Sun Apr  8 20:31:34 CDT 2018 (#9268)

- v2.0 in PX4/Firmware (3ec37aa186a07681e6383f1cefa565a1ee74eaef): https://github.com/mavlink/c_library_v2/commit/95d4cb3b21130d67b251f323482f1c3fe1ce0c6c
    - v2.0 current upstream: https://github.com/mavlink/c_library_v2/commit/163d4c84b9eb58a02cab94d039f54c303977d991
    - Changes: https://github.com/mavlink/c_library_v2/compare/95d4cb3b21130d67b251f323482f1c3fe1ce0c6c...163d4c84b9eb58a02cab94d039f54c303977d991

    163d4c8 2018-04-08 PX4BuildBot - autogenerated headers for rev https://github.com/mavlink/mavlink/tree/07c4fc5f617383707317128e87ec9a0489ed3634

dagar committed:
tests temporarily disable mavlink tests on nuttx

dagar committed:
tests add ctlmath and use float FLT_EPSILON

dagar committed:
fix typo PX4IO_MAX_MIXER_LENGHT -> PX4IO_MAX_MIXER_LENGTH

dagar committed:
px4fmu_test update mixers

dagar committed:
NuttX cleanup debug helpers and add profile target

acfloria committed:
Move the SimpleAnalyzer to a separate file

bkueng committed:
voted_sensors_update: reduce flash usage by allowing compiler to dedup equal strings

reduces roughly 130 bytes. Almost enough to pass CI.

bkueng committed:
ASSERT: remove some inappropriate asserts

Since assertions lead to crashes, we need better failure handling. In all
the cases in this patch, the assert is not required.

All the ones with the task id should be replaced with the module base
class.

Ah yes, and this reduces flash space, since the ASSERT macro will expand to
a printf that contains the source file name.

bkueng committed:
nuttx_px4fmu-v2_test.cmake: enable modules required to build mc_pos_control

due to param dependencies

acfloria committed:
Move HIGH_LATENCY2 stream to separate file and update some comments"

acfloria committed:
MavlinkReceiver: Don't start MissionManager in IRIDIUM mode

dagar committed:
px4iofirmware use std NAN instead of undefined 0.0f/0.0f

 - closes #9277

dagar committed:
mixer lib remove custom constrain and cleanup includes

dagar committed:
stop manually defining physical constants

MaEtUgR committed:
mc_att_control: improve a quaternion control comment

acfloria committed:
Mavlink: Don't use a packed struct as input for the convert_limit_safe function

- Addional small cleanup of the iridium driver and the include guards of mavlink module header files

potaito committed:
tap_esc: made _uart_fd a class member

potaito committed:
tap_esc: moved common functions to tap_esc_common

potaito committed:
tap_esc_common: docs & function return types etc.

bkueng committed:
commander: set timestamp for vehicle_command publication

potaito committed:
tap_esc: removed unused mode var/struct

mhkabir committed:
mavlink : fix pointer for main_mode and sub_mode

dagar committed:
enable -fno-math-errno as we never check errno (#9281)

- disables setting of the errno variable as required by C89/C99 on calling math library routines

dagar committed:
EKF2 move all orb_subscribe/unsubsribe to the constructor/destructor

dagar committed:
commander move telemetry checks to new preflightcheck helper

potaito committed:
tap_esc: removed unnecessary includes

bkueng committed:
logger: remove some unneeded SITL topics

We got over the limit

potaito committed:
tap_esc: removed unused NO_VERIFY_CONFIG

potaito committed:
tap_esc: removed redundant _initialized member

potaito committed:
tap_esc: more cleanup
- removing unused variables
- removing unused code blocks
- improving code readability
- initialising members at definition

dagar committed:
Jenkins add OSX builds and disable circleci (#9198)

dagar committed:
move check stack to cmake and run in Jenkins

bkueng committed:
drv_hrt: add user-defined integer literals for time constants

The goal is to improve the readability for expressions like:
if (hrt_elapsed_time(&start_time) > 8000000)

Available since C++11

bkueng committed:
land_detector: use user-defined literals for time constants

potaito committed:
mag_calibration: only allocate as much memory as needed

TSC21 committed:
modules: mavlink: drop globallocalconverter in ODOMETRY rcv

Commit:
Update submodule sitl_gazebo to latest Fri Apr 13 20:26:27 CDT 2018

    - sitl_gazebo in PX4/Firmware (4ffa98aa2a1418c9eab5bf72ec6f0e93ef2f9860): https://github.com/PX4/sitl_gazebo/commit/28921bca9df9a212675b8e888c926d1ec6f6b28e
    - sitl_gazebo current upstream: https://github.com/PX4/sitl_gazebo/commit/0df46c53b14ef4241b33fb7a74a3cb6f9d83c6dd
    - Changes: https://github.com/PX4/sitl_gazebo/compare/28921bca9df9a212675b8e888c926d1ec6f6b28e...0df46c53b14ef4241b33fb7a74a3cb6f9d83c6dd

    0df46c5 2018-04-13 Tully Foote - Only use catkin env hook command if building with catkin.
38c0d07 2018-04-12 Tully Foote - Add package to GAZEBO_RESOURCE_PATH to make the worlds accessible.
5811488 2018-04-11 Tully Foote - updating OpticalFlow submodule
7c79bd7 2018-04-10 Tully Foote - Add environment hooks to export paths for plugins and resources.

Commit:
Update submodule DriverFramework to latest Fri Apr 13 20:26:36 CDT 2018

    - DriverFramework in PX4/Firmware (1297859680ce15980a970c95d7caa95be67957c7): https://github.com/PX4/DriverFramework/commit/f98ea65e9bd35a8d2bdedd39e519b7320fe82b16
    - DriverFramework current upstream: https://github.com/PX4/DriverFramework/commit/e06222dd151e2f92476762e483915df5bd67929d
    - Changes: https://github.com/PX4/DriverFramework/compare/f98ea65e9bd35a8d2bdedd39e519b7320fe82b16...e06222dd151e2f92476762e483915df5bd67929d

    e06222d 2018-04-11 Daniel Agar - cmake change px4_add_module -> px4_add_library

dagar committed:
Update submodule v2.0 to latest Fri Apr 13 20:26:31 CDT 2018

- v2.0 in PX4/Firmware (23017d051b7ff6686c8c8ff025a8e98af094780a): https://github.com/mavlink/c_library_v2/commit/163d4c84b9eb58a02cab94d039f54c303977d991
    - v2.0 current upstream: https://github.com/mavlink/c_library_v2/commit/c55dd0ec312a667f28b0cab4f1fe697d2888c00c
    - Changes: https://github.com/mavlink/c_library_v2/compare/163d4c84b9eb58a02cab94d039f54c303977d991...c55dd0ec312a667f28b0cab4f1fe697d2888c00c

    c55dd0e 2018-04-12 PX4BuildBot - autogenerated headers for rev https://github.com/mavlink/mavlink/tree/292874beed6bf756be45421bd2ae2420de921f2b
e2916e5 2018-04-10 PX4BuildBot - autogenerated headers for rev https://github.com/mavlink/mavlink/tree/012765b923546224ef928caeb86c72d5a78a433a

dagar committed:
Jenkins add tailsitter CI mission

dagar committed:
Jenkins set CCACHE_BASEDIR on Mac

dagar committed:
Update submodule ecl to latest Fri Apr 13 20:26:41 CDT 2018 (#9303)

- ecl in PX4/Firmware (163fdded13d1cb48ea4849a4596feaa7025cb796): https://github.com/PX4/ecl/commit/ba2b9dfdd96d50d697165407b88b5bc94cdef84c
    - ecl current upstream: https://github.com/PX4/ecl/commit/02e319431b948f660f8548707849489d6c822fad
    - Changes: https://github.com/PX4/ecl/compare/ba2b9dfdd96d50d697165407b88b5bc94cdef84c...02e319431b948f660f8548707849489d6c822fad

02e3194 2018-04-11 Daniel Agar - cmake remove embedded vim settings
91b5bfe 2018-04-11 Daniel Agar - attitude_fw remove unused perf include
02055ac 2018-04-09 Paul Riseborough - EKF: Fix non GPS aiding data reset logic (#418)

TSC21 committed:
replace att_pos_mocap uORB topics with *_groundtruth topics

TSC21 committed:
modules: mavlink: add ODOMETRY msg sender

TSC21 committed:
modules: mavlink: fix how the states and covariances are propagated

TSC21 committed:
modules: mavlink: remove LOCAL_POSITION_NED_COV rcv

TSC21 committed:
modules: mavlink: convert from quaternion to euler covariances

TSC21 committed:
modules: mavlink: remove ATTITUDE_QUATERNION_COV rcv

TSC21 committed:
modules: mavlink: remove accel bias covariance parsing on LPNC msg

TSC21 committed:
modules: mavlink: add ODOMETRY frame_id and child_frame_id

Commit:
mavlink: HIGHRES_IMU set diff_pressure in fields_updated (#9296)

TSC21 committed:
msg: tools: rtps: add new IDs without breaking compatibility

hamishwillee committed:
Fix description of VT_FW_ALT_ERR

bkueng committed:
rcS: do not reset COM_FLTMODE* on airframe change

An airframe change keeps all RC params except the flight modes. This fixes
that.

potaito committed:
tap_esc: addressing review comments for #9313

potaito committed:
tap_esc: formatting and renaming

potaito committed:
tap_esc: moved device path to drv_tap_esc.h

potaito committed:
tap_esc: paranoid check

potaito committed:
tap_esc: replaced PWM_DEFAULT_UPDATE_RATE

potaito committed:
rc_check: typos in mavlink error

bkueng committed:
module template: move documentation to a new category 'template' (#9324)

dagar committed:
EKF2 fix airspeed orb_copy check

RomanBapst committed:
FixedWingPositionControl: rotate attitude consistenly for tailsitters

Signed-off-by: Roman <bapstroman@gmail.com>

TSC21 committed:
modules: mavlink: receiver: update the way the data is handled according with the frame

RomanBapst committed:
tailsitter: use forward transition throttle for climb-rate controlled mode

- the mc pos controller will decrease throttle during the transition
and thus the vehicle will not pick up enough airspeed to complete the
transition

Signed-off-by: Roman <bapstroman@gmail.com>

RomanBapst committed:
posix-configs: better tailsitter front transition parameters

- reduce height increase during front transition

Signed-off-by: Roman <bapstroman@gmail.com>

RomanBapst committed:
posix-configs: set tailsitter SITL SYS_AUTOSTART to quad tailsitter

Signed-off-by: Roman <bapstroman@gmail.com>

RomanBapst committed:
posix-configs: increase tailsitter land speed

Signed-off-by: Roman <bapstroman@gmail.com>

priseborough committed:
ecl : update to version that fixes wind relative dead reckoning bug

dagar committed:
Jenkins add VTOL tiltrotor mission test

potaito committed:
tap_esc: removed a zero-memset

potaito committed:
libecl: checked out recent commit of master (#9339)

 * ECL update includes physical constants for https://github.com/PX4/Firmware/pull/9319

RomanBapst committed:
mpu9250 wrapper: support updating gyro & accel cutoff via parameter

- useful for tuning snapdragon flight

Signed-off-by: Roman <bapstroman@gmail.com>

TSC21 committed:
modules: mavlink: use Dcm to apply transforms between body<->local frames

TSC21 committed:
sync submodules

TSC21 committed:
modules: mavlink: apply the transforms differently to linear and angular velocities

dagar committed:
FW landing check if prev valid before using (#9284)

dagar committed:
Update submodule nuttx to latest px4_firmware_nuttx-7.22+ (#9354)

    - Changes: https://github.com/PX4-NuttX/nuttx/compare/fe0dc6c41a6bd7d942945459ac960ff2d3e1f4fc...65b87d092a6d5f776ab5f9dbb095d31e45861789

    65b87d0 2018-04-19 Daniel Agar - [PX4 BACKPORT] Merged in dagar/nuttx/pr-stm32f777_typo (pull request #628)

bkueng committed:
mc_att_control_params: increase max acro rates to 1800

This matches the maximum rates for the attitude controller.

bkueng committed:
MC acro: add separate params to configure yaw expo

So that it can be customized better according to personal preferences.

The limitation of 16 chars did not allow to use a better param name.

bkueng committed:
MC: set default MPC_THR_MAX & MPC_MANTHR_MAX to 1

With the updated mixer (#9062) it's safe to set maximum thrust to 1 (in
both cases, if MC_AIRMODE is set, or not set).
So I see no reason to limit the maximum thrust.

bkueng committed:
mixer_multirotor: fix initialization of _outputs_prev

memset (http://www.cplusplus.com/reference/cstring/memset/) interprets
the value argument as unsigned char, so passing a float will not work.

bkueng committed:
mc_att_control_main: do not apply the yaw weight for the feedforward term

The meaning of the yaw weight changed with #8003:
- before, the yaw weight decreased with increasing tilt angle error, so
  it was mostly 1
- now, it is constant and depends on the tuning gains (around 0.4 by default)

It means that #8003 reduced the feedforward term, and we get the closer
behavior as before with this change.
It also reduces coupling between different parameters.

dagar committed:
commander use new time literals

ChristophTobler commented on src/gazebo_vision_plugin.cpp:
```diff
@@ -167,12 +194,46 @@ void VisionPlugin::OnUpdate(const common::UpdateInfo&)
+    gazebo::msgs::Vector3d* position = new gazebo::msgs::Vector3d();
+    position->set_x(pose_model.Pos().X() + noise_pos.X() + _bias.X());
+    position->set_x(pose_model.Pos().Y() + noise_pos.Y() + _bias.Y());
+    position->set_x(pose_model.Pos().Z() + noise_pos.Z() + _bias.Z());
```
`set_x()` -> `set_y()` and `set_z()`

TSC21 replied:
Yeah. Typo. Thanks.

ChristophTobler reviewed (changes requested):
This doesn't seem to be tested... The gazebo error is resolved by the comment but still, if I run this PR with `EKF2_AID_MASK = 24` and `EKF2_HGT_MODE = 3` the vehicle immediately flies away after takeoff.

bkueng committed:
imu drivers: do not set on-chip filter based on driver filter setting

The description of IMU_GYRO_CUTOFF was incorrectly saying that it only
affects the driver filtering, but in fact it also set the on-chip filter
to the next higher supported value. This patch fixes that.

And because the IMU_GYRO_CUTOFF and not the IMU_ACCEL_CUTOFF was used for
the on-chip filter, after #9070 which sets the default gyro filter to 80,
we were effectively using a dlpf of 98 Hz.
For this reason this patch changes the on-chip cutoff frequency to 98 as
well, so that the overall default behavior is unchanged.

TSC21 committed:
fix rebase issues
